registrations/models.py

```diff
@@ -347,24 +347,44 @@ def can_be_edited_by(self, user):
     def is_user_editable_resources(self):
         return bool(self.data_source and self.data_source.user_editable_resources)
 
+    def get_service_language_pk(self):
+        if self.service_language:
+            return self.service_language.pk
+        return "fi"
+
     def send_notification(self, confirmation_type):
+        service_language = self.get_service_language_pk()
+
         email_variables = {
             "linked_events_ui_url": settings.LINKED_EVENTS_UI_URL,
             "linked_registrations_ui_url": settings.LINKED_REGISTRATIONS_UI_URL,
             "username": self.name,
-            "event": self.registration.event.name_fi,
+            "event": getattr(
+                self.registration.event,
+                f"name_{service_language}",
+                self.registration.event.name,
+            ),
```

Comment:
Unless there is a specific reason why not, it would be cleaner to use django-modeltranslation's standard way of accessing the translated fields, ie.
```
with translation.override(service_language):
    email_variables = {
        ...
        "event": self.registration.event
        ...
    }
```
One thing to consider is what kind of fallback language logic we want to use with these fields, and [configure modeltranslation](https://django-modeltranslation.readthedocs.io/en/latest/usage.html#fallback-languages) accordingly.